core/src/main/java/org/hisp/dhis/android/core/fileresource/internal/FileResourceStorageStatusVerifier.kt

```diff
@@ -0,0 +1,182 @@
+
+        val results = mutableMapOf<String, FileResourceVerificationResult>()
+        val pendingUids = fileResourceUids.toMutableSet()
+
+        for (attempt in 1..maxAttempts) {
+            if (pendingUids.isEmpty()) {
+                break
+            }
+
+            Log.d(TAG, "Verification attempt $attempt/$maxAttempts for ${pendingUids.size} pending files")
+
+            // Process in batches to avoid overwhelming the server
+            val currentBatch = pendingUids.take(batchSize)
+            val batchResults = verifyBatch(currentBatch)
+
+            batchResults.forEach { (uid, result) ->
+                when (result.status) {
+                    FileResourceStorageStatus.STORED -> {
+                        results[uid] = result
+                        pendingUids.remove(uid)
+                        Log.d(TAG, "File resource $uid is STORED")
+                    }
+                    FileResourceStorageStatus.FAILED -> {
+                        results[uid] = result
+                        pendingUids.remove(uid)
+                        Log.w(TAG, "File resource $uid FAILED to store")
+                    }
+                    FileResourceStorageStatus.PENDING -> {
+                        // Keep in pending list for next attempt
+                        Log.d(TAG, "File resource $uid still PENDING")
+                    }
+                    FileResourceStorageStatus.NONE -> {
+                        // Treat as pending
+                        Log.d(TAG, "File resource $uid has status NONE")
+                    }
+                }
+            }
+
+            // If there are still pending files and we haven't reached max attempts, wait before next poll
+            if (pendingUids.isNotEmpty() && attempt < maxAttempts) {
+                delay((delayMs * (2.0.pow(attempt) - 1)).toLong())
+            }
+        }
+
+        // Mark remaining pending files as timed out
+        pendingUids.forEach { uid ->
+            results[uid] = FileResourceVerificationResult(
+                uid = uid,
+                status = FileResourceStorageStatus.PENDING,
+                isVerified = false,
+                timedOut = true,
+            )
+            Log.w(TAG, "File resource $uid timed out after $maxAttempts attempts")
+        }
+
+        val storedCount = results.count { it.value.isVerified }
+        val failedCount = results.count { it.value.status == FileResourceStorageStatus.FAILED }
+        val timedOutCount = results.count { it.value.timedOut }
+
+        Log.i(
+            TAG,
+            "Batch verification complete: $storedCount stored, $failedCount failed, $timedOutCount timed out",
+        )
+
+        return results
+    }
+
+    /**
+     * Verifies a single batch of file resources.
+     */
+    private suspend fun verifyBatch(uids: List<String>): Map<String, FileResourceVerificationResult> {
+        return uids.associateWith { uid ->
```

Comment:
I think the `verifyBatch` processes files sequentially rather than in parallel. With each verification taking 1-2 seconds, a batch of 10 files takes 10-20 seconds instead of a couple of seconds.

Should we try something like this?

```
private suspend fun verifyBatch(uids: List<String>): Map<String, FileResourceVerificationResult> {
    return coroutineScope {
        uids.map { uid ->
            async { uid to verifyFileResource(uid) }
        }.awaitAll().toMap()
    }
}
```